File: scripts/check-multi.py
import re
data = open('shared/concepts.js').read()
entries = re.findall(r"'([^']+)':\{href:withHash\('([^']+)'", data)
print(f"Total entries: {len(entries)}")

from collections import defaultdict
by_url = defaultdict(list)
for name, url in entries:
    by_url[url].append(name)

multi = {url:names for url,names in by_url.items() if len(names) > 1}
print(f"Multi-concept lessons: {len(multi)}")
for url,names in sorted(multi.items(), key=lambda x:-len(x[1])):
    print(f"  {', '.join(names)}  -> {url}")

File: scripts/check-multi.js
// Check multi-concept lesson mappings
const fs = require("fs");
const data = fs.readFileSync("shared/concepts.js","utf-8");
const match = data.match(/const lessonByConcept = \{([\s\S]*?)\};/);
const block = match[1];
const lines = block.split("\n").filter(l => l.includes("href:withHash"));
const entries = lines.map(l => {
  const nm = l.match(/^\s+'([^']+)'/);
  const hf = l.match(/withHash\('([^']+)'/);
  return {name: nm?.[1] ?? "?", href: hf?.[1] ?? "?"};
});
console.log("Total entries: " + entries.length);
const byUrl = {};
entries.forEach(e => {
  if (!byUrl[e.href]) byUrl[e.href] = [];
  byUrl[e.href].push(e.name);
});
const multi = Object.entries(byUrl).filter(([_,n]) => n.length > 1);
console.log("Multi-concept lessons: " + multi.length);
multi.sort((a,b) => b[1].length - a[1].length).forEach(([url,names]) => console.log("  " + names.join(", ") + " -> " + url));
